lib/awful/screenshot.lua

```diff
@@ -0,0 +1,884 @@
+---------------------------------------------------------------------------
+--- Screenshots and related configuration settings
+--
+-- @author Brian Sobulefsky &lt;brian.sobulefsky@protonmail.com&gt;
+-- @copyright 2021 Brian Sobulefsky
+-- @inputmodule awful.screenshot
+---------------------------------------------------------------------------
+
+-- Grab environment we need
+local capi = { root = root,
+               screen = screen,
+               client = client,
+               mousegrabber = mousegrabber }
+local aw_screen = require("awful.screen")
+local gears = require("gears")
+local beautiful = require("beautiful")
+local wibox = require("wibox")
+local cairo = require("lgi").cairo
+local naughty = require("naughty")
+
+-- The module to be returned
+local module = { mt = {} }
+-- The screenshow object created by a call to the screenshow module
+local screenshot = {}
+-- The various methods of taking a screenshow (root window, client, etc).
+local screenshot_methods = {}
+
+-- Configuration data
+local ss_dir = nil
+local ss_prfx = nil
+local frame_color = gears.color("#000000")
+local initialized = nil
+
+
+-- Internal function to get the default screenshot directory
+-- 
+-- Can be expanded to read X environment variables or configuration files.
+local function get_default_dir()
+  -- This can be expanded
+  local d = os.getenv("HOME")
+  
+  if d then
+    d = string.gsub(d, '/*$', '/') .. 'Images/'
+    if os.execute("bash -c \"if [ -d \\\"" .. d .. "\\\" -a -w \\\"" .. d ..
```

Comment:
..speaking of which, - this should go - again, see gears.filesystem for helper function for working with filesystem

Reply:
I'll look at gears.filesystem and see what is available.